fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepaymentScheduleInstallment.java

```diff
@@ -625,7 +626,7 @@ public Money writeOffOutstandingPenaltyCharges(final LocalDate transactionDate,
     }
 
     public boolean isOverdueOn(final LocalDate date) {
-        return getDueDate().isBefore(date);
+        return DateUtils.isAfter(date, getDueDate());
```

Comment:
Arent these changes irrelevant to [FINERACT-1921](https://issues.apache.org/jira/browse/FINERACT-1921)? Please extract them into a separate PR if so!